File: MovieCatalogQw/app/src/main/java/com/unlistedi/moviecatalogqw/DetailActivity.java
package com.unlistedi.moviecatalogqw;

import androidx.appcompat.app.AppCompatActivity;

import android.media.Image;
import android.os.Bundle;
import android.widget.Adapter;
import android.widget.GridView;
import android.widget.ImageView;
import android.widget.TextView;

import com.unlistedi.moviecatalogqw.data.MovieEntity;
import com.unlistedi.moviecatalogqw.utils.GenreAdapter;

public class DetailActivity extends AppCompatActivity {

    public static final String EXTRA_MOVIE = "extra_movie";

    private GenreAdapter genreAdapter;
    private TextView tvDetailTitle, tvDetailScore, tvDetailYear, tvDetailDescription;
    private ImageView ivDetailPoster;
    private MovieEntity movieEntity;
    private GridView gvGenre;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        movieEntity = getIntent().getParcelableExtra(EXTRA_MOVIE);
        if(movieEntity != null){
            genreAdapter = new GenreAdapter(this, movieEntity.getGenre());

            gvGenre = findViewById(R.id.gvGenre);
            tvDetailTitle = findViewById(R.id.tvDetailTitle);
            tvDetailScore = findViewById(R.id.tvDetailScore);
            tvDetailYear = findViewById(R.id.tvDetailYear);
            tvDetailDescription = findViewById(R.id.tvDetailDescription);
            ivDetailPoster = findViewById(R.id.ivDetailPoster);

            gvGenre.setAdapter(genreAdapter);
            tvDetailTitle.setText(movieEntity.getTitle());
            tvDetailScore.setText(String.valueOf(movieEntity.getScore()));
            tvDetailYear.setText(movieEntity.getYear());
            tvDetailDescription.setText(movieEntity.getDescription());
            ivDetailPoster.setImageResource(movieEntity.getImg_path());
            gvGenre.setAdapter(genreAdapter);
        }

    }
}

File: MovieCatalogueKu/app/src/main/java/com/unlistedi/moviecatalogueku/DetailActivity.kt
package com.unlistedi.moviecatalogueku

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.BaseAdapter
import android.widget.TextView
import com.unlistedi.moviecatalogueku.data.MovieEntity
import com.unlistedi.moviecatalogueku.utils.GenreAdapter
import kotlinx.android.synthetic.main.activity_detail.*
import kotlinx.android.synthetic.main.items_genre.*

class DetailActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_MOVIE = "movie_detail"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val movieEntity = intent.getParcelableExtra<MovieEntity>(EXTRA_MOVIE)

        if (movieEntity != null){
            val genreAdapter = GenreAdapter(movieEntity.genre, this)
            gvGenre.adapter = genreAdapter

            tvDetailTitle.text = movieEntity.title
            ivDetailPoster.setImageResource(movieEntity.img_path)
            tvDetailScore.text = movieEntity.score.toString() + "%"
            tvDetailYear.text = movieEntity.year
            tvDetailDescription.text = movieEntity.description
        }

    }

}

